Enforce a specific TLS version

jeffdesc opened the pull request:
This PR adds the possibility to use a very specific TLS version instead of using the auto negotiation that normally happens.

The way this feature is implemented should not break any existing parameters and will need all parameters set if the TLS version is required to be set.

Documentation is updated as required.

Please let me know if any changes need to be made.

selivan commented:
Hello!

Thank you for the contribution. I really appreciate that you updated not just the script, but the docs as well 👍 

Also thank you for mentioning https://badssl.com, I will update README to have actually working examples, not fake ones.

The line
> See "man s_client" for supported TLS versions

can be a little confusing, because "man s_client" does not explicitly mention TLS versions.

I suggest referencing "TLS Version Options" section of [man openssl](https://www.openssl.org/docs/man3.0/man1/openssl.html) and just taking this options without leading dash: `tls1`, `tls1_2`. This way:

* the script will support not just TLS N.N versions, but all of openssl protocol version options, like `no_tls1_3`, `dtls` (I just learned that thing exists).
* script parameters and zabbix items will look a little more descriptive: `tls_1_3` is self-descriptive, `1.3` could be some numerical parameter

jeffdesc commented:
Hey @selivan 

Thanks for the correct feedback. Glad we could learn both in this PR!

I have updated the script to:
* fully take the argument in (`tls1`, `tls1_3`, `dtls1`, ...) without replacing characters now
* only accept `tls`, `ssl` and `dtls` as argument (but also allow the `no_tls...` formats), otherwise make it empty

The documentation is also updated to reflect both places of where they can find the valid TLS options.
